tests/utils_tests/test_numberformat.py

```diff
@@ -19,6 +19,12 @@ def test_format_number(self):
         self.assertEqual(nformat(1234, '.', grouping=2, thousand_sep=',',
                                  force_grouping=True), '12,34')
         self.assertEqual(nformat(-1234.33, '.', decimal_pos=1), '-1234.3')
+        self.assertEqual(nformat(10000, '.', grouping=3,
```

Comment:
Please format similar to 575706331bec4bf58ce36a9540c4c61fca49025b to avoid non-4 space indent. I guess these are just additional test cases not affected by the optimizations so they would probably be better as a separate commit to make that clear.